src/main/java/org/myrobotlab/framework/Service.java

```diff
@@ -1700,6 +1705,15 @@ public void sendToPeer(String peerName, String method, Object... data) {
   }
 
   public void send(String name, String method, Object... data) {
+    // if you know the service is local - use same thread
+    // to call directly
+    ServiceInterface si = Runtime.getService(name);
```

Comment:
i'm consistently getting the arduino unit test stuck...  i suspect its related to this change